### PR TITLE
3ds Max OSL blend shader

### PR DESCRIPTION
**Pull request by usakhelo**
Hi,

This PR is complement to [3ds max plugin PR](https://github.com/appleseedhq/appleseed-max/pull/82)
Adds OSL blend shader and adapt max disney shader to work it.

Thanks,
Sergo.

### CODE REVIEW

File: src/appleseed.shaders/CMakeLists.txt [hunk 1]
```diff
@@ -306,6 +306,7 @@ set (src_max_sources
      src/max/as_max_float_texture.osl
      src/max/as_max_normal_map.osl
      src/max/as_max_srgb_to_linear_rgb.osl
+	 src/max/as_max_blend_material.osl
```
Comment (dictoon) [inline src/appleseed.shaders/CMakeLists.txt]: Remove tab.

**Comment by usakhelo** [comment]
Hi,
MixColor is never textured, it's just a color. Mask color is a grayscale texture.
I tried to mimic vray blend material, but it's not exactly like this. It's mix color is overridden by texture if texture amount is 100%. In my case both mix amount and texture contribute equally.

**Comment by est77** [comment]
I think that the PR good. We'll merge after the 1.8 release.

Thanks!

File: src/appleseed.shaders/CMakeLists.txt [hunk 2]
```diff
@@ -306,6 +306,7 @@ set (src_max_sources
      src/max/as_max_float_texture.osl
      src/max/as_max_normal_map.osl
      src/max/as_max_srgb_to_linear_rgb.osl
+     src/max/as_max_blend_material.osl
```
Comment (dictoon) [inline src/appleseed.shaders/CMakeLists.txt]: Order alphabetically.

File: src/appleseed.shaders/src/max/as_max_blend_material.osl [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+    output	closure color ClosureOut = 0
+)
+{
+    ClosureOut = BaseMtl;
+    
+    if (LayerMtl_1)
+        ClosureOut = mix(ClosureOut, LayerMtl_1, (MixColor_1 * MaskColor_1));
+    if (LayerMtl_2)
+        ClosureOut = mix(ClosureOut, LayerMtl_2, (MixColor_2 * MaskColor_2));
```
Comment (dictoon) [inline src/appleseed.shaders/src/max/as_max_blend_material.osl]: Is this the right mixing algorithm? Or more precisely, the mixing algorithm we actually want?

Here, the weight of `LayerMtl_1` will be

    (MixColor_1 * MaskColor_1) * (1 - MixColor_2 * MaskColor_2)

It's likely that this is what we want. Just checking.

File: src/appleseed.shaders/CMakeLists.txt [hunk 4]
```diff
@@ -301,6 +301,7 @@ source_group ("src\\gaffer\\vector" FILES
 
 set (src_max_sources
      src/max/as_max_bump_map.osl
+     src/max/as_max_blend_material.osl
```
Comment (dictoon) [inline src/appleseed.shaders/CMakeLists.txt]: Not quite ordered yet but it's ok :)

**Comment by dictoon** [comment]
Merged, thanks!

File: src/appleseed.shaders/CMakeLists.txt (hunk 4 above)
Comment (usakhelo) [reply src/appleseed.shaders/CMakeLists.txt]: That's shame, sorry!

**Pull request merged**